File: src/javascript/LargestNumber.js
const largestNumber = function (nums) {
  nums.sort((a, b) => {
    let sa = a.toString()
    let sb = b.toString()
    return Number.parseInt(sa + sb) > Number.parseInt(sb + sa) ? -1 : 1
  })
  if (nums[0] === 0) {
    return '0'
  }

  return nums.join('')
};
const nums = [10, 2];
console.log(largestNumber(nums));

File: src/LargestNumber.java
import java.util.Arrays;

public class LargestNumber {

  public static void main(String[] args) {
    var nums = new int[]{10, 2};
    System.out.println(largestNumber(nums));
  }

  public static String largestNumber(int[] nums) {
    String[] s = new String[nums.length];
    for (int i = 0; i < nums.length; i++) {
      s[i] = String.valueOf(nums[i]);
    }
    Arrays.sort(s, (a, b) -> {
      String o1 = a + b;
      String o2 = b + a;
      return o2.compareTo(o1);
    });
    return s[0].equals("0") ? "0" : String.join("", s);
  }
}
